src/vision/frc_vision_final.py

```diff
@@ -0,0 +1,190 @@
+# Contains most of the vision processing, though the most complete is under Offline_Filter
+# Likely to become the final product file for vision processing
+# The exposure options are likely needed to be adjusted per competition
+# They are currently tuned to the testing area
+# Tuning for the RealSense camera can be done easily through the ReaslSense Viewer app in a GUI
+# One thing I do still need to figure out is assigning the camera to a specific IO port number
+# Testing on my laptop, it is only visible when using one port
+
+from math import degrees, radians
+import pyrealsense2 as rs2
+import cv2
+import numpy as np
+import time
+from networktables import NetworkTables
+
+ANGLETHRESHOLD = 40
+
+# Takes in slopes x and y, tests if they are equal to each other or any previously verified line
+
+
+def unequal(new, old_list):
+    variance = 5
+    for i in old_list:
+        x3, y3, x4, y4 = i[0]
+        old_slope = degrees(np.arctan((y4 - y3)/(x4 - x3)))
+        min_val = old_slope - variance
+        max_val = old_slope + variance
+        if min_val < new < max_val:
```

Comment:
`if abs(new - old_slop) < variance` is shorter, and you can get rid of the above two lines